ocpi-toolkit-2.2.1/src/main/kotlin/com/izivia/ocpi/toolkit/modules/versions/VersionsClient.kt

```diff
@@ -39,6 +35,18 @@ class VersionsClient(
                         allowTokenA = true
                     )
             )
-                .parseBody()
+                .also {
+                    if (it.status != HttpStatus.OK) throw HttpException(it.status, parseHttpStatus(it.status.code).name)
+                }
+                .runCatching {
+                    this.parseBody<OcpiResponseBody<List<Version>>>()
+                }
+                .onFailure {
+                    throw OcpiServerUnusableApiException(
+                        "Could not get versions from server, there was an error parsing the response: " +
+                            "URL='$serverVersionsEndpointUrl', error='${it.message}'"
+                    )
```

Comment:
I think `OcpiServer*` exceptions should only be used on server side. Here it's the client, and there was an error in the body returned by the partner server.

So we should use an OcpiResponseException:

```kotlin
OcpiResponseException(
    statusCode = CLIENT_ERROR.code,
    statusMessage = "Could not get versions from server, there was an error parsing the response: " +
                        "URL='$serverVersionsEndpointUrl', error='${it.message}'"
)
```

Or a new OcpiResponseException error that explicitely says that there was an error parsing the response of the server. `OcpiResponseParsingException` that inherits from `OcpiResponseException` or something like that.

This exception does not have to follow OCPI spec as it will only be read by the person that uses the lib. So it has to be clear for the person that uses the lib

Reply:
You're right, the exception is not a server exception, and doesn't need to follow the OCPI specification.

I'm thinking about the new exception may be something like `OcpiToolkitResponseParsingException` inside `OcpiToolkitExceptions.tk`

What do you think?

Reply:
Yes, I think it's a good approach. We will probably need new ocpi toolkit exceptions in the future